### PR TITLE
Throw `not_impl_error` for `approx_percentile_cont` parameters validation 

### LINKED ISSUE #12012: Internal error in `approx_percentile_cont()` aggregate function (SQLancer)
### Describe the bug

Reproducer:
1. Run datafusion-cli under latest main branch
```sh
(base) arrow-datafusion/datafusion-cli|main🌝| >>> git pull public main
From https://github.com/apache/arrow-datafusion
 * branch                main       -> FETCH_HEAD
Already up to date.
(base) arrow-datafusion/datafusion-cli|main🌝| >>> git checkout 2a16704db

...

(base) arrow-datafusion/datafusion-cli|2a16704db🌝| >>> cargo run
```
2. Execute queries
```
DataFusion CLI v41.0.0
> create table t1(v1 double);
0 row(s) fetched.
Elapsed 0.078 seconds.

> select approx_percentile_cont(v1,v1) from t1;
Internal error: PhysicalExpr Column references column 'v1' at index 0 (zero-based) but input schema only has 0 columns: [].
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

I think this query is invalid, the second argument have to be a literal number instead of column reference, to make this query meaningful.
However an internal error should not be thrown, it always represent a potential bug in DataFusion.
So we have to further investigate if it's a actual bug or misused internal error

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

Found by SQLancer https://github.com/apache/datafusion/issues/11030

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

Closes #12012 .

## What changes are included in this PR?
If the `percentile` or `t-digest` is invalid for `approx_percentile_cont`, throw `no_impl_err` with a readable message instead of `internal_error`.

## Are these changes tested?
yes

## Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/functions-aggregate/src/approx_percentile_cont.rs [hunk 1]
```diff
@@ -154,19 +154,20 @@ fn get_scalar_value(expr: &Arc<dyn PhysicalExpr>) -> Result<ScalarValue> {
 }
 
 fn validate_input_percentile_expr(expr: &Arc<dyn PhysicalExpr>) -> Result<f64> {
-    let percentile = match get_scalar_value(expr)? {
-        ScalarValue::Float32(Some(value)) => {
+    let percentile = match get_scalar_value(expr).ok() {
```
Comment (alamb) [inline datafusion/functions-aggregate/src/approx_percentile_cont.rs:157]: Another way to express this for your consideration that might make the error handling more obvious is

```rust
      let scalar = match get_scalar_value(expr)
        .map_err(not_impl_err!("Percentile value for 'APPROX_PERCENTILE_CONT' must be a litera, got: {expr}l")?
      match scalar { 
          ScalarValue::Float32(Some(value)) => {
 ...
 ```

**Review by alamb — APPROVED** [review]
Thanks @goldmedal  -- this looks like an improvement to me

I had another suggestion on how to simplify the code, but it is certainly not required

File: datafusion/functions-aggregate/src/approx_percentile_cont.rs (hunk 1 above)
Comment (goldmedal) [reply datafusion/functions-aggregate/src/approx_percentile_cont.rs]: It looks nice. Thanks for the suggestions!

**Comment by jayzhan211** [comment]
Thanks @goldmedal @alamb

**Comment by goldmedal** [comment]
Thanks @alamb @jayzhan211

**Pull request merged**